File: 3423-maximum-difference-between-adjacent-elements-in-a-circular-array/3423-maximum-difference-between-adjacent-elements-in-a-circular-array.py
class Solution:
    def maxAdjacentDistance(self, nums: List[int]) -> int:
        n = len(nums)
        max = abs(nums[0] - nums[n-1])

        for i in range(1,n):
            if max < abs(nums[i] - nums[i-1]):
                max = abs(nums[i] - nums[i-1])
        
        return max

File: 3423-maximum-difference-between-adjacent-elements-in-a-circular-array/3423-maximum-difference-between-adjacent-elements-in-a-circular-array.java
class Solution {
    public int maxAdjacentDistance(int[] nums) {
        
        int n = nums.length;
        int max =Math.abs(nums[n-1] - nums[0]);
        for(int i=0; i<n-1; i++){
            max = Math.max(max, Math.abs(nums[i+1] - nums[i]));
        }
        return max;
    }
}